File: docs/sdks/typescript/examples/insan-iq.ts
import { Lydian } from '@lydian/sdk';

const lydian = new Lydian({
  apiKey: process.env.LYDIAN_API_KEY,
});

async function insanIQExample() {
  // 1. Create a doctor persona
  const persona = await lydian.insanIQ.createPersona({
    name: 'Dr. Sarah Chen',
    type: 'doctor',
    description: 'Cardiology specialist with 15 years of experience',
    capabilities: [
      'diagnosis',
      'treatment-planning',
      'patient-consultation',
      'research',
    ],
    metadata: {
      specialty: 'Cardiology',
      languages: ['English', 'Mandarin'],
      certifications: ['ABIM', 'FACC'],
    },
  });

  console.log('Persona created:', persona.id);

  // 2. Publish skills to persona
  const skill = await lydian.insanIQ.publishSkill({
    personaId: persona.id,
    name: 'Echocardiogram Analysis',
    category: 'diagnostic',
    proficiencyLevel: 95,
    description: 'Expert in reading and interpreting echocardiograms',
    metadata: {
      yearsOfExperience: 12,
      casesAnalyzed: 5000,
    },
  });

  console.log('Skill published:', skill.id);

  // 3. List all skills
  const skills = await lydian.insanIQ.listSkills(persona.id);
  console.log(`Persona has ${skills.data.length} skills`);

  // 4. Create chat session
  const session = await lydian.insanIQ.createSession({
    personaId: persona.id,
    userId: 'user-123',
    title: 'Patient consultation - Case #456',
  });

  console.log('Session created:', session.id);

  // 5. Send messages
  const message1 = await lydian.insanIQ.sendMessage(
    session.id,
    'Patient presents with chest pain and shortness of breath'
  );

  console.log('User message:', message1.content);

  // Get assistant response
  await new Promise(resolve => setTimeout(resolve, 1000)); // Wait for processing

  // 6. Get session history
  const history = await lydian.insanIQ.getSessionHistory(session.id);
  console.log(`Session has ${history.data.length} messages`);

  history.data.forEach(msg => {
    console.log(`[${msg.role}]: ${msg.content.substring(0, 100)}...`);
  });
}

insanIQExample().catch(console.error);

File: Desktop/ailydian-ultra-pro/docs/sdks/python/examples/insan_iq.py
"""İnsan IQ API examples"""

import os
from lydian import Lydian

lydian = Lydian(api_key=os.environ.get("LYDIAN_API_KEY"))

# 1. Create persona
persona = lydian.insan_iq.create_persona(
    name="Dr. Sarah Chen",
    persona_type="doctor",
    description="Cardiology specialist with 15 years of experience",
    capabilities=["diagnosis", "treatment-planning", "patient-consultation"],
    metadata={"specialty": "Cardiology", "languages": ["English", "Mandarin"]}
)
print(f"Persona created: {persona.id}")

# 2. Publish skill
skill = lydian.insan_iq.publish_skill(
    persona_id=persona.id,
    name="Echocardiogram Analysis",
    category="diagnostic",
    proficiency_level=95,
    description="Expert in reading and interpreting echocardiograms",
    metadata={"years_of_experience": 12, "cases_analyzed": 5000}
)
print(f"Skill published: {skill.id}")

# 3. List skills
skills = lydian.insan_iq.list_skills(persona.id)
print(f"Persona has {len(skills.data)} skills")

# 4. Create chat session
session = lydian.insan_iq.create_session(
    persona_id=persona.id,
    user_id="user-123",
    title="Patient consultation - Case #456"
)
print(f"Session created: {session.id}")

# 5. Send message
message = lydian.insan_iq.send_message(
    session.id,
    "Patient presents with chest pain and shortness of breath"
)
print(f"Message sent: {message.id}")

# 6. Get session history
history = lydian.insan_iq.get_session_history(session.id)
print(f"Session has {len(history.data)} messages")

for msg in history.data:
    print(f"[{msg.role}]: {msg.content[:100]}...")
